Compilation failure due to unused results in mock dnsmasq

With GCC 8.3.

```
/home/ubuntu/mydev/multipass/multipass/tests/mock_dnsmasq_server.cpp: In function ‘int main(int, char
**)’:
/home/ubuntu/mydev/multipass/multipass/tests/mock_dnsmasq_server.cpp:38:9: error: ignoring return value of ‘int pipe(int*)’, declared with attribute warn_unused_result [-Werror=unused-result]
     pipe(pipefd);
     ~~~~^~~~~~~~
/home/ubuntu/mydev/multipass/multipass/tests/mock_dnsmasq_server.cpp:59:22: error: ignoring return value of ‘ssize_t write(int, const void*, size_t)’, declared with attribute warn_unused_result [-Werror=unused-result]
                 write(pipefd[1], "1", 1);
                 ~~~~~^~~~~~~~~~~~~~~~~~~
/home/ubuntu/mydev/multipass/multipass/tests/mock_dnsmasq_server.cpp:63:18: error: ignoring return value of ‘ssize_t write(int, const void*, size_t)’, declared with attribute warn_unused_result [-Werror=unused-result]
             write(pipefd[1], "0", 1);
             ~~~~~^~~~~~~~~~~~~~~~~~~
/home/ubuntu/mydev/multipass/multipass/tests/mock_dnsmasq_server.cpp:71:13: error: ignoring return value of ‘ssize_t read(int, void*, size_t)’, declared with attribute warn_unused_result [-Werror=unused-result]
         read(pipefd[0], &exit_code, 1);
         ~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

--- a/tests/mock_dnsmasq_server.cpp
+++ b/tests/mock_dnsmasq_server.cpp
@@ -19,8 +19,12 @@
 #include <QCoreApplication>
 #include <QFile>
 
-#include <sys/types.h>
+#include <cerrno>
+#include <cstring>
+#include <iostream>
 #include <unistd.h>
+
+const auto unexpected_error = 5;
 
 int main(int argc, char* argv[])
 {
@@ -35,7 +39,11 @@ int main(int argc, char* argv[])
     parser.parse(QCoreApplication::arguments());
 
     int pipefd[2];
-    pipe(pipefd);
+    if (pipe(pipefd))
+    {
+        std::cerr << "Failed to create pipe: " << std::strerror(errno) << std::endl;
+        return unexpected_error;
+    }
 
     char exit_code;
 
@@ -54,21 +62,29 @@ int main(int argc, char* argv[])
         if (parser.isSet(listenOption))
         {
             auto address = parser.value(listenOption);
-            if (address.contains("0.0.0"))
+            auto contains000 = address.contains("0.0.0");
+
+            if (write(pipefd[1], contains000 ? "1" : "0", 1) < 1)
             {
-                write(pipefd[1], "1", 1);
-                return 1;
+                std::cerr << "Failed to write to pipe: " << std::strerror(errno) << std::endl;
+                return unexpected_error;
             }
 
-            write(pipefd[1], "0", 1);
+            if (contains000)
+                return 1;
         }
+
         return QCoreApplication::exec();
     }
     else if (pid > 0)
     {
         close(pipefd[1]);
 
-        read(pipefd[0], &exit_code, 1);
+        if (read(pipefd[0], &exit_code, 1) < 1)
+        {
+            std::cerr << "Failed to read from pipe: " << std::strerror(errno) << std::endl;
+            return unexpected_error;
+        }
 
         return static_cast<int>(exit_code);
     }

